src/xcode/ENA/ENA/Source/Scenes/Home/HomeInteractor.swift

```diff
@@ -427,3 +431,30 @@ extension HomeInteractor {
 		homeViewController.showExposureNotificationSetting()
 	}
 }
+
+// MARK: - CountdownTimerDelegate methods.
+
+/// The `CountdownTimerDelegate` is used to update the remaining time that is shown on the risk cell button until a manual refresh is allowed.
+extension HomeInteractor: CountdownTimerDelegate {
+	private func scheduleCountdownTimer() {
+		let nextUpdate = riskProvider.nextExposureDetectionDate()
+		countdownTimer = CountdownTimer(countdownTo: nextUpdate)
+		countdownTimer?.delegate = self
+		countdownTimer?.start()
+	}
+
+	func update(time: String) {
+		guard let indexPath = self.indexPathForRiskCell() else { return }
+		guard let cell = homeViewController.cellForItem(at: indexPath) as? RiskLevelCollectionViewCell else { return }
+
+		// We pass the time and let the configurator decide whether the button can be activated or not.
+		riskLevelConfigurator?.timeUntilUpdate = time
+		riskLevelConfigurator?.configureButton(for: cell)
+	}
+
+	func done() {
```

Comment:
IMO, it's better to use the Apple style to name the delegate method, for example,
```swift
 func  countdownTimer(_ timer: CountdownTimer, didUpdate time: String)
```

Reply:
I agree, I will adjust that!